tools/export/cces/__init__.py

```diff
@@ -0,0 +1,418 @@
+        Converts the compiler flags into CCES options and removes them
+        from the flags list
+        """
+        options  = {}
+
+        enable_optimization = "true"
+        value = "arm.base.compiler.option.optimization.og"
+        for flag in flags:
+            if flag.startswith("-O"):
+                value = "arm.base.compiler.option.optimization.o" + flag[2:]
+                if flag[2:] == "0":
+                    enable_optimization = "false"
+                flags.remove(flag)
+                break
+        option = Option("baseId", value)
+        options["arm.base.compiler.option.optimization"] = option
+
+        option = Option("baseId", enable_optimization)
+        options["arm.base.compiler.option.optimization.enable"] = option
+
+        booleans = {"-g": "arm.base.compiler.option.debug",
+                    "-save-temps": "arm.base.compiler.option.savetemps",
+                    "-ffunction-sections": "arm.c.compiler.option.elimination.code",
+                    "-fdata-sections": "arm.c.compiler.option.elimination.data",
+                    "-pedantic": "arm.base.compiler.option.pedantic",
+                    "-pedantic-errors": "arm.base.compiler.option.pedanticerrors",
+                    "-w": "arm.base.compiler.option.inhibitallwarnings",
+                    "-Wall": "arm.base.compiler.option.allwarnings",
+                    "-Wextra": "arm.base.compiler.option.extrawarnings",
+                    "-Werror": "arm.base.compiler.option.warningaserror",
+                    "-Wconversion": "arm.base.compiler.option.conversionwarning"}
+
+        for flag in booleans:
+            value = "false"
+            if flag in flags:
+                value = "true"
+                flags.remove(flag)
+            option = Option("baseId", value)
+            options[booleans[flag]] = option
+
+        CCES.convert_common_options("arm.base.compiler.", options, flags)
+
+        return options
+
+    @staticmethod
+    def convert_linker_options(flags):
+        """
+        Converts the linker flags into CCES options and removes them
+        from the flags list
+        """
+        options = {}
+
+        booleans = {"-nostartfiles": "arm.linker.option.nostart",
+                    "-nodefaultlibs": "arm.linker.option.nodeflibs",
+                    "-nostdlib": "arm.linker.option.nostdlibs",
+                    "-s": "arm.linker.option.strip",
+                    "-Wl,--gc-sections": "arm.linker.option.elimination"}
+
+        for flag in booleans:
+            value = "false"
+            if flag in flags:
+                value = "true"
+                flags.remove(flag)
+            option = Option("baseId", value)
+            options[booleans[flag]] = option
+
+        return options
+
+    # override
+    def generate(self):
+        """
+        Generate the CCES project files using headless builder.
+        """
+
+        self.resources.win_to_unix()
+
+        t = TARGET_MAP[self.target]
```

Comment:
You should not create your own target object. Please use `toolchain.target` instead.